pkg/infra/remotecache/remotecache_test.go

```diff
@@ -0,0 +1,108 @@
+package remotecache
```

Comment:
These tests looks a little odd compared to the rest of Grafana but I liked the idea of reusing the same tests for all cache client implementations. 

I liked `"github.com/bmizerany/assert"` (its already vendored) since it makes assert short and precise. 
I think we should use something similar for smaller and tests cases like this and only use convey for scenario testing.